fpmsyncd/fpmsyncd.cpp

```diff
@@ -174,6 +197,30 @@ int main(int argc, char **argv)
                     {
                         s.removeSelectable(&eoiuCheckTimer);
                     }
+                }          
+                else if (temps == &reconcileCheckTimer)
+                {
+                    SWSS_LOG_NOTICE("reconcile Check timer Expired ");
+                    if (sync.m_warmStartHelper.inProgress())
+                    {
+                        if (sync.isReadyToReconcile())
+                        {
+                            reconcileHoldTimer.setInterval(timespec{2, 0});
```

Comment:
What is the hold-timer for?  Why do we need wait another 2 seconds before reconcile happens?

Reply:
Once fpmsyncd reconcile is done, this timer checks if orchagent is reconciled and we are ready to start updating the reconciled fpmsyncd entries into the APP-DB. Since did not want to check too frequently, kept the timer at 2 seconds.

Reply:
reconcileHoldTimer is one shot timer which is fired once the control plane reconciliation is done. At this time if orchagnet is reconciled, we continue with fpnsyncd reconcillation. However when reconcileHoldTimer expires if orchangent is still not reconciled, we fire reconcileHoldTimer for another 2 seconds and re-check if the orchagent is reconciled yet.  This continues till it finds orchagent has reconciled. The idea is it wait for minimum of control plane ( BGP  ) reconciliation time ( default 120 seconds) and also check if orchagent is reconciled before reconciling fpmsyncd. Both conditions should be met. Orchagent reconcile can occur earlier or later based on system scale, however we need to wait for minimum control plane  ( BGP ) reconcillation time, before reconciling fpmsyncd.

Reply:
Some correction in the above explanation, reconcileCheckTimer ( not reconcileHoldTimer) is fired again for 2 seconds to re-check if orchagent has reconciled. Hence reconcileHoldTimer does not need 2 seconds. Will change it to 1 seconds like eoiu hold timer.